src/NSwag.Annotations/OpenApiCallbackAttribute.cs

```diff
@@ -0,0 +1,104 @@
+        /// <param name="callbackUrl"></param>
+        public OpenApiCallbackAttribute(string callbackUrl) : this(callbackUrl, name: null, type: null)
+        { }
+        
+        /// <summary>Initializes a new instance of the <see cref="OpenApiCallbackAttribute"/> class.</summary>
+        /// <param name="name"></param>
+        /// <param name="callbackUrl"></param>
+        /// <param name="method"></param>
+        public OpenApiCallbackAttribute(string callbackUrl,
+                                        string name = null,
+                                        string method = "post") : this(callbackUrl, name, method, mimeType : null, type: null)
+        { }
+                
+
+        /// <summary>Initializes a new instance of the <see cref="OpenApiCallbackAttribute"/> class.</summary>
+        /// <param name="callbackUrl"></param>
+        /// <param name="types"></param>
+        public OpenApiCallbackAttribute(string callbackUrl,
+                                        params Type[] types) : this(callbackUrl, name: null, types: types)
+        { }
+
+        /// <summary>Initializes a new instance of the <see cref="OpenApiCallbackAttribute"/> class.</summary>
+        /// <param name="name"></param>
+        /// <param name="callbackUrl"></param>
+        /// <param name="method"></param>
+        /// <param name="mimeType"></param>
+        /// <param name="type"></param>
+        public OpenApiCallbackAttribute(string callbackUrl,
+                                        string name = null,
+                                        string method = "post",
+                                        string mimeType = null,
+                                        Type type = null) : this(callbackUrl, name, method, mimeType, type != null ? new[] { type } : new Type[] { })
+        { }
+
+
+        /// <summary>Initializes a new instance of the <see cref="OpenApiCallbackAttribute"/> class.</summary>
+        /// <param name="name"></param>
+        /// <param name="callbackUrl"></param>
+        /// <param name="method"></param>
+        /// <param name="mimeType"></param>
+        /// <param name="types"></param>
+        public OpenApiCallbackAttribute(string callbackUrl,
+                                        string name = null, 
+                                        string method = "post",
+                                        string mimeType = null,
+                                        params Type[] types) : base()
+        {
+            if (string.IsNullOrWhiteSpace(name))
+            {
+                throw new ArgumentException("Invalid callback name", nameof(name));
+            }
+
+            this.Name = name;
+            this.Types = types;
+            this.CallbackUrl = callbackUrl;
+            this.Method = method;
+            this.MimeType = mimeType;
+        }
+        /// <summary>
+        /// The name of the callback
+        /// </summary>
+        public string Name { get; }
```

Comment:
I think in the other attributes we also have setters to change properties (not only in the ctor). Maybe we should do it the same way in this attribute...